src/main/java/org/sunbird/operationalreports/service/OperationalReportServiceImpl.java

```diff
@@ -359,14 +376,45 @@ private Map<String, Object> getSearchObject(List<String> userIds) {
         return requestObject;
     }
 
-    private long getParsedDate(String reportExpiryDate) throws ParseException {
-        long reportExpiryDateMillis;
-        SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd");
-        Date parsedDate = dateFormat.parse(reportExpiryDate);
-        reportExpiryDateMillis = parsedDate.getTime();
-        return reportExpiryDateMillis;
+    public SBApiResponse readGrantAccess(String authToken,String userOrgId) {
+        SBApiResponse response = ProjectUtil.createDefaultResponse(Constants.READ_REPORT_ACCESS_API);
+        Map<String, Object> primaryKeyMap = new HashMap<>();
+        primaryKeyMap.put(Constants.USER_ORG_ID,userOrgId);
+        Map<String, Object> userAccessReportExpiryDetails = getReportAccessDetails(userOrgId);
+        List<Map<String, Object>> result = new ArrayList<>();
+        for(Map.Entry<String, Object> entry:userAccessReportExpiryDetails.entrySet()){
+            Map<String, Object> mdoIdAccessDetails = new HashMap<>();
+            mdoIdAccessDetails.put(Constants.USER_ID,entry.getKey());
+            mdoIdAccessDetails.put(Constants.REPORT_ACCESS_EXPIRY, ((Map<String, Object>) entry.getValue()).get(Constants.USER_REPORT_EXPIRY_DATE));
+            result.add(mdoIdAccessDetails);
+        }
+        response.getResult().put(Constants.STATUS, Constants.SUCCESS);
+        response.getResult().put(Constants.RESPONSE, result);
+        return response;
+    }
+
+    private Map<String, Object> getReportAccessDetails(String userOrgId) {
+        Map<String, Object> primaryKeyMap = new HashMap<>();
+        primaryKeyMap.put(Constants.USER_ORG_ID, userOrgId);
+        return cassandraOperation.getRecordsByPropertiesByKey(Constants.KEYSPACE_SUNBIRD,
```

Comment:
use the cassandraOperation.***WithoutAllowFiltering -- methods should be used...